mm2src/coins/lightning/ln_storage.rs

```diff
@@ -0,0 +1,143 @@
+use super::*;
+use common::mm_ctx::MmArc;
+use lightning::routing::network_graph::NetworkGraph;
+use lightning::routing::scoring::Scorer;
+use lightning::util::ser::{Readable, Writeable};
+use secp256k1::PublicKey;
+use std::collections::HashMap;
+use std::fs::{File, OpenOptions};
+use std::io::{BufReader, BufWriter};
+use std::net::{SocketAddr, ToSocketAddrs};
+use std::path::{Path, PathBuf};
+use std::str::FromStr;
+
+pub fn my_ln_data_dir(ctx: &MmArc, ticker: &str) -> PathBuf { ctx.dbdir().join("LIGHTNING").join(ticker) }
+
+pub fn my_ln_data_backup_dir(path: String, ticker: &str) -> PathBuf {
+    PathBuf::from(path).join("LIGHTNING").join(ticker)
+}
+
+pub fn nodes_data_path(ctx: &MmArc, ticker: &str) -> PathBuf { my_ln_data_dir(ctx, ticker).join("channel_nodes_data") }
+
+fn nodes_data_backup_path(path: String, ticker: &str) -> PathBuf {
+    my_ln_data_backup_dir(path, ticker).join("channel_nodes_data")
+}
+
+pub fn network_graph_path(ctx: &MmArc, ticker: &str) -> PathBuf { my_ln_data_dir(ctx, ticker).join("network_graph") }
+
+pub fn scorer_path(ctx: &MmArc, ticker: &str) -> PathBuf { my_ln_data_dir(ctx, ticker).join("scorer") }
+
+fn pubkey_and_addr_from_str(pubkey_str: &str, addr_str: &str) -> ConnectToNodeResult<(PublicKey, SocketAddr)> {
```

Comment:
I guess as this module is called `ln_storage`, its functions should not return `ConnectToNodeResult` - it is used in another context. Also, please create the `LnStorage` trait and implement it for a new `FileStorage` struct similarly to `coins::my_tx_history_v2::TxHistoryStorage`.

Reply:
Done. I implemented the new `Storage` trait for the already used `struct FileSystemPersister` instead of creating a new `FileStorage` struct